File: 1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts/1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts.cpp
class Solution {
public:
    int maxArea(int H, int W, vector<int>& h, vector<int>& v)
    {
        int n1 = h.size();
        int n2 = v.size();
        sort(h.begin() , h.end());
        sort(v.begin() , v.end());
        int mod = 1e9+7;
        
        int a = -1 , b = -1;
        
        a = h[0];
        b = v[0];
        
        for(int i=0;i<n1-1;i++)
        {
            a = max(a , (int)(h[i+1]-h[i]));
        }
        
        a = max(a , H - h[n1-1] );
        
        for(int i=0;i<n2-1;i++)
        {
            b = max(b , v[i+1]-v[i]);
        }
        
        b = max(b , W - v[n2-1]);
        a = a%mod;
        b = b%mod;
        return (1LL*a*b)%mod;
        
    }
};

File: 1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts/1465-maximum-area-of-a-piece-of-cake-after-horizontal-and-vertical-cuts.java
class Solution {
    int MOD = (int)1e9 + 7;
    
    public int maxArea(int h, int w, int[] hc, int[] vc) {
        int n = hc.length , m = vc.length;
        
        int []hc1 = new int[n + 2] , vc1 = new int[m + 2];
        
        hc1[0] = vc1[0] = 0; hc1[n + 1] = h; vc1[m + 1] = w ;
        
        for(int i = 0 ; i < n ; i++){
            hc1[i + 1] = hc[i];
        }
        
        for(int i = 0; i < m ; i++){
            vc1[i + 1] = vc[i];
        }
        
        Arrays.sort(hc1) ; Arrays.sort(vc1);
        // System.out.println(Arrays.toString(hc1)  + " " + Arrays.toString(vc1));
        int max1 = -1 , max2 = -1;
        
        for(int i = 0 ; i < hc1.length - 1 ; i++){
            max1 = Math.max(max1 , hc1[i + 1] - hc1[i]);
        }
        
        for(int i = 0 ; i < vc1.length - 1 ; i++){
            max2 = Math.max(max2 , vc1[i + 1] - vc1[i]);
        }
        
        return (int)(((long)(max1%MOD)*(max2%MOD))%(long)MOD);
    }
}